### PR TITLE
fix #195891: score information too big for small screens

### PR DESCRIPTION
**Pull request by lasconic**

### CODE REVIEW

**Comment by jeetee** [comment]
No luck on Win7 with your patch yet on a first testrun, I'll have another go after factory reset to see if that helps

**Comment by jeetee** [comment]
No dice, after factory reset:

`Warning: setGeometryDp: Unable to set geometry 615x513+493+174 on QWidgetWindow/'UploadScoreDialogWindow'. Resulting geometry:  615x782+493+174 (frame: 8, 30, 8, 8, custom margin: 0, 0, 0, 0, minimum size: 474x513, maximum size: 16777215x16777215). (:0, )`

File: mscore/uploadscoredialog.ui [hunk 1]
```diff
@@ -7,7 +7,7 @@
     <x>0</x>
     <y>0</y>
     <width>615</width>
-    <height>644</height>
+    <height>438</height>
```
Comment (Jojo-Schmitz) [inline mscore/uploadscoredialog.ui]: Here QtCreator doesn't allow to go lower than 510

**Comment by Jojo-Schmitz** [comment]
Those changes to master and the dialog window is 623x750. Just 6 pixels less than an x765 Screen. Exactly the same height without the change...

**Comment by lasconic** [comment]
Sorry, my commit was not complete... What about now?

**Comment by Jojo-Schmitz** [comment]
Yes! That brings it down to 507 pixels minimum height (again without the upload ssore audio part though)

**Comment by jeetee** [comment]
Without custom audio the warning now is, not able to set 473, while that also is its recorded minimal size; the resulting height is then 486.
With custom audio, the warning is "not able to set 513" (again, 513 as minimal height) resulting in 526 actual height.

Both are again well within the boundaries of a 768 screen, so I'd consider this fixed.
Note that we are trading the offscreen bug for a possible "textarea's don't grow if more space is available" feature request. Still an improvement imo.

File: mscore/uploadscoredialog.ui [hunk 2]
```diff
@@ -129,6 +126,9 @@
          <property name="locale">
           <locale language="English" country="UnitedStates"/>
          </property>
+         <property name="sizeAdjustPolicy">
+          <enum>QAbstractScrollArea::AdjustToContentsOnFirstShow</enum>
```
Comment (Jojo-Schmitz) [inline mscore/uploadscoredialog.ui]: Maybe this and the other could get set programatically, only if the screen is not high enough?

Comment (Jojo-Schmitz) [inline mscore/uploadscoredialog.ui]: Maybe this and the other could get set programatically, only if the screen is not high enough?
And this here and or the other (or the other only), only if the other is also visible, i.e. on updating an existing score?

Actually I've tried it out (it is possible programatically): both settings are needed on screens less than 786 pixels high, only the other setting on screens less than 800 pixels high.

File: mscore/uploadscoredialog.cpp [hunk 3]
```diff
@@ -69,6 +69,7 @@ UploadScoreDialog::UploadScoreDialog(LoginManager* loginManager)
                            .arg("</a>"));
       QFont font = licenseHelp->font();
       font.setPointSize(8);
+      font.setItalic(true);
       licenseHelp->setFont(font);
```
Comment (Jojo-Schmitz) [inline mscore/uploadscoredialog.cpp]: Not unsetting italic again? Same font is used later again, and there it was not italic before this change.

Comment (lasconic) [inline mscore/uploadscoredialog.cpp]: Yes, now it is, and all "comments" or help messages are italic. Look a bit better.

File: mscore/uploadscoredialog.cpp [hunk 4]
```diff
@@ -78,6 +79,7 @@ UploadScoreDialog::UploadScoreDialog(LoginManager* loginManager)
 
       tagsHelp->setText(tr("Use a comma to separate the tags"));
```
Comment (Jojo-Schmitz) [inline mscore/uploadscoredialog.cpp]: This text is a duplicate, same on the UI file, so needs to get translated twice, I think

Comment (lasconic) [inline mscore/uploadscoredialog.cpp]: No, there is a notr=true in the UI file. And so all the help messages are in C++

Comment (Jojo-Schmitz) [inline mscore/uploadscoredialog.cpp]: OK, must have missed that

File: mscore/uploadscoredialog.cpp (hunk 3 above)
Comment (Jojo-Schmitz) [inline mscore/uploadscoredialog.cpp]: Fine by me, in my PR I just tried to stay as close as possible with the old layout

**Pull request merged**